crates/astria-conductor/src/executor.rs

```diff
@@ -196,6 +218,19 @@ impl<C: ExecutionClient> Executor<C> {
             );
             return Ok(Some(execution_hash.clone()));
         }
+        // if the block that we just recieved is not the next block in the sequence to be executed,
+        // add it to the block queue
+        if let Some(parent_block) = block.header.last_block_id {
```

Comment:
Should we refactor a bit here? 

The `execute_block` logic feels pretty fat. We first early escape if not ready to execute, then process, and execute. There's an oddity to `execute_block` adding to the queue and `execute_queue` calling back here. Duplication of logic in validating block is ready to execute which if done incorrectly, could result in a the cycle of adding block back onto queue and then never executing. Done right, no issues but two places to mess up increases chances.

My thinking is that upon receiving a new block we should:
- process it down to what needs to be executed, fail fast log if issues here
- push the processed block to the queue (minimal data in the queue)

There can then be a separate async loop which checks the top of the queue and sees if it is ready to be executed, and executes. This can run pretty quickly, just iterating on the process of executing from queue.

cc @SuperFluffy for thoughts

Reply:
i moved the queue execution logic into `execute_block`. this is messy atm but I made #219 to clean up `execdute_block`.

Reply:
I agree, this is a very complicated function and I don't quite grok what's going on here. Let's refactor it here instead of leaving it to a followup.

Reply:
Will do!